portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java

```diff
@@ -7014,18 +7025,38 @@ protected void validate(
 			Locale locale)
 		throws PortalException {
 
+		validate(
+			companyId, -1, userId, autoPassword, password1, password2,
+			autoScreenName, screenName, emailAddress, openId, firstName,
+			middleName, lastName, organizationIds, locale);
+	}
+
+	protected void validate(
+			long companyId, long ldapServerId, long userId,
+			boolean autoPassword, String password1, String password2,
+			boolean autoScreenName, String screenName, String emailAddress,
+			String openId, String firstName, String middleName, String lastName,
+			long[] organizationIds, Locale locale)
+		throws PortalException {
+
 		validateMaxUsers(companyId);
 
 		if (!autoScreenName) {
 			validateScreenName(companyId, userId, screenName);
 		}
 
 		if (!autoPassword) {
-			PasswordPolicy passwordPolicy =
-				_passwordPolicyLocalService.getDefaultPasswordPolicy(companyId);
+			PasswordPolicy passwordPolicy = null;
 
-			PwdToolkitUtil.validate(
-				companyId, 0, password1, password2, passwordPolicy);
+			if ((ldapServerId <= -1) ||
+				!LDAPSettingsUtil.isPasswordPolicyEnabled(companyId)) {
+
+				passwordPolicy =
+					_passwordPolicyLocalService.getDefaultPasswordPolicy(
+						companyId);
+			}
```

Comment:
Shouldn't this use `ldapServerId <= 0` as condition? Or is this a special case? In all other places I've seen, it is either `ldapServerId > 0` or its complement `ldapServerId <= 0`.

Reply:
It shouldn't functionally matter, since the only users with an ldapServerId of 0 (default user, anonymous user) won't reach this condition anyways.  I will change it back to <= 0 though, for consistency.